util/src/test/java/io/kubernetes/client/pager/PagerTest.java

```diff
@@ -0,0 +1,90 @@
+package io.kubernetes.client.pager;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static org.junit.Assert.assertEquals;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import com.google.common.io.Resources;
+import io.kubernetes.client.ApiClient;
+import io.kubernetes.client.apis.CoreV1Api;
+import io.kubernetes.client.models.V1Namespace;
+import io.kubernetes.client.models.V1NamespaceList;
+import io.kubernetes.client.util.ClientBuilder;
+import io.kubernetes.client.util.PagerParams;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class PagerTest {
+
+  private ApiClient client;
+  private static final String LIST_FILE_PATH =
+      Resources.getResource("namespace-list.json").getPath();
+  private static final int PORT = 8087;
+  @Rule public WireMockRule wireMockRule = new WireMockRule(PORT);
+
+  @Before
+  public void setup() throws IOException {
+    client = new ClientBuilder().setBasePath("http://localhost:" + PORT).build();
+  }
+
+  @Test
```

Comment:
Let's add/expand the tests so that they actually cover the pagination.

Reply:
+1. add let's add test when the list receives a `400` due to invalid `limit` or `continue`, maybe expecting a checked exception here?

Reply:
added 2 more tests for failure cases.